src/DynamicData/Binding/BindAndSort.cs

```diff
@@ -0,0 +1,160 @@
+                }
+            });
+
+    private void Reset(IEnumerable<TObject> sorted)
+    {
+        if (target is ObservableCollectionExtended<TObject> observableCollectionExtended)
+        {
+            using (observableCollectionExtended.SuspendNotifications())
+            {
+                observableCollectionExtended.Load(sorted);
+            }
+        }
+        else
+        {
+            target.Clear();
+            foreach (var t in sorted)
+            {
+                target.Add(t);
+            }
+        }
+    }
+
+    private void ApplyChanges(IChangeSet<TObject, TKey> changes)
+    {
+        // iterate through collection, find sorted position and apply changes
+
+        foreach (var change in changes.ToConcreteType())
+        {
+            var item = change.Current;
+
+            switch (change.Reason)
+            {
+                case ChangeReason.Add:
+                    {
+                        var index = GetInsertPosition(item);
+                        target.Insert(index, item);
+                    }
+                    break;
+                case ChangeReason.Update:
+                    {
+                        var currentIndex = GetCurrentPosition(change.Previous.Value);
+                        target.RemoveAt(currentIndex);
+
+                        var index = GetInsertPosition(item);
+                        target.Insert(index, item);
+                    }
+                    break;
+                case ChangeReason.Remove:
+                    {
+                        var currentIndex = GetCurrentPosition(item);
+                        target.RemoveAt(currentIndex);
+                    }
+                    break;
+                case ChangeReason.Refresh:
+                    {
+                        var currentIndex = target.IndexOf(item);
+                        var index = GetInsertPosition(item);
+
+                        // this assumption may be dodgy as we may need to remove the original item first
+                        if (index != currentIndex)
+                        {
+                            target.RemoveAt(currentIndex);
+                            target.Insert(index, item);
+                        }
+                    }
+                    break;
+                case ChangeReason.Moved:
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+    }
+
+    private int GetCurrentPosition(TObject item)
+    {
+        var index = options.UseBinarySearch ? target.BinarySearch(item, comparer) : target.IndexOf(item);
+
+        if (index < 0)
+        {
+            throw new SortException($"Cannot find item: {typeof(TObject).Name} -> {item} from {target.Count} items");
+        }
+
+        return index;
+    }
+
+    private int GetInsertPosition(TObject item) => options.UseBinarySearch ? GetInsertPositionBinary(item) : GetInsertPositionLinear(item);
```

Comment:
Probably, this is a meaningless micro-optimization, but what I would do is either separate this logic into different subclasses, or make this a `Func<TObject, int>` field that gets assigned to one method or the other, upon subscription. That way, the if check here isn't getting repeated for every inserted item, it only gets done once, since the options don't ever change.

Reply:
I think it's probably a micro-optimization too far as a simple bool check is inexpensive.